specification/security_model.adoc

```diff
@@ -1,9 +1,323 @@
+assigned TDIs, on a per-TVM basis.
+
+Each TDI must be exclusively assigned to no more than one TVM. However, a single
+TVM can simultaneously be assigned several TDIs, irrespective of whether they
+originate from the same physical device or not.
+
+As a TVM TCB does not include PCIe switches and bridges, a selective PCIe IDE
+stream must be setup to guarantee end-to-end confidentiality and integrity
+protection between a TVM and a TDI. The host VMM is responsible for setting up a
+single selective IDE stream between the host platform and any physical device
+from which one or more TDIs are assigned to TVMs. All TDIs within that physical
+device share the single PCIe IDE stream.
+
+The main security objective of the CoVE-IO security model is to protect a TVM's
+confidential data integrity and confidentiality while TDIs are assigned to it.
+At the same time, availability of those assigned TDIs is out of this model scope
+as e.g. the host VMM could remove them from the TVM at any time it sees fit.
+Either the DSM or the TDI itself must clear and wipe all TVM confidential data
+in the TDI before the host software stack can fully reclaim an assigned TDI.
+
+== Threat Model
+
+=== Assets
+
+The CoVE-IO security model aims at protecting the following assets:
+
+1. The TVM confidential data, which includes its confidential main memory, code
+   and execution state.
+2. An assigned TDI trusted I/O space that is mapped into the TVM address space,
+   also known as a TDI trusted MMIO.
 
 === Adversary Model
 
-=== Scope and Threat Model
+The CoVE-IO security model aims at protecting the above-described TVM assets
+from the following adversaries:
+
+* _Privileged host software adversary_: This includes host software components
+  executing in S and M mode like the host firmware, kernel, hypervisor, VMM,
+  etc. As the system resource owner but also the TVMs lifecycle manager, those
+  components can access and control all devices on the system.
+
+* _Unprivileged host software adversary_: This includes host software components
+  executing in U mode like e.g. the userspace parts of the host VMM.
+
+* _Privileged guest software adversary_: This includes guest software components
+  executing in VS mode like e.g. the guest kernel.
+
+* _Unprivileged guest software adversary_: This includes guest software components
+  executing in VU mode like e.g. the guest application workload.
+
+* _Device firmware adversary_: This includes any firmware driving a device
+  within the system's PCIe topology. As PCIe transaction generators those device
+  firmware components can gain direct access to a TVM confidential memory.
+
+* _Hardware adversary_: This includes adversaries that have unlimited physical
+  access to the system and its devices, and can use simple or advanced hardware
+  attacks such as bus interposers, voltage/clock glitchers, microscopes, power
+  usage analyzer, etc. in order to access or tamper with the protected assets.
```

Comment:
maybe better to split HW adversary into simple HW adversary and advanced HW adversary

Reply:
Done.